src/actions/wallet/writeUnsafeDepositTransaction.ts

```diff
@@ -24,7 +24,8 @@ type DepositTransactionParameters = {
 export type WriteUnsafeDepositTransactionParameters<
   TAbi extends Abi | readonly unknown[] = typeof optimismPortalABI,
   TFunctionName extends string = 'depositTransaction',
-  TChain extends Chain | undefined = Chain,
+  TToChain extends OpChainL2 = OpChainL2,
+  TChain extends Chain & { id: TToChain['l1']['id'] } = TToChain["l1"],
```

Comment:
```suggestion
  TFromChain extends Chain & { id: TToChain['l1']['id'] } = TToChain["l1"],
```

Reply:
hmm I find this naming a little odd for some reason. I kinda like keeping the main client chain with this same generic name as viem. But it also makes sense in some ways and is clear.